### PR TITLE
Add tolerations and taints to Snapshot statefulset, remove ^M

### PR DESCRIPTION
**Pull request by Quarky9**
**Is this a bug fix or adding new feature?**
Add same tolerations and taints to snapshot set as the main controller

**What is this PR about? / Why do we need it?**
Allow deployment of optional control along main via helm chart

**What testing is done?** 
Deployed private test cluster using forked helm chart

### CODE REVIEW

File: aws-ebs-csi-driver/templates/statefulset.yaml [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
-      serviceAccount: ebs-snapshot-controller
-      containers:
-        - name: snapshot-controller
-          image: quay.io/k8scsi/snapshot-controller:v2.0.1
-          args:
-            - --v=5
-            - --leader-election=false
-{{- end }}
+{{- if .Values.enableVolumeSnapshot }}
+#Snapshot controller
```
Comment (msau42) [inline aws-ebs-csi-driver/templates/statefulset.yaml]: The snapshot controller should be deployed as part of Kubernetes cluster deployment, not the CSI driver. The snapshot controller handles all operations across all CSI drivers.

Comment (Quarky9) [reply aws-ebs-csi-driver/templates/statefulset.yaml]: Hi,
I understand that the snapshot controller can be used for various CSI implementation within the same cluster, but in terms of placement within the cluster:

How is that different from the main ebs-csi-controller ?
Both are part of the extended control plane of each cluster and we run them on the controller nodes.
Thx for any clarification ;-)

Comment (msau42) [reply aws-ebs-csi-driver/templates/statefulset.yaml]: For example, let's say you deploy all these csi drivers in your cluster: ebs, efs, lustre, ceph, xyz, etc.

Even though you have 1+ csi driver in the cluster, only 1 snapshot-controller should be deployed. Therefore snapshot controller deployment needs to live outside of any one csi driver.

Comment (leakingtapan) [reply aws-ebs-csi-driver/templates/statefulset.yaml]: What @msau42 says makes sense if that how snapshot controller was designed. The situation now is there is no such kubernetes cluster deployment creates the snapshot controller on aws, and we need a place to create the controller

Comment (leakingtapan) [reply aws-ebs-csi-driver/templates/statefulset.yaml]: It will be great if kops can create it after cluster creation./cc @justinsb

Comment (msau42) [reply aws-ebs-csi-driver/templates/statefulset.yaml]: In the meantime, can we at least make the snapshot controller deployment optional with clear documentation that you should only have one deployed per cluster?

Comment (msau42) [reply aws-ebs-csi-driver/templates/statefulset.yaml]: I would also caution against putting the snapshot CRD definitions in any sort of automated driver deployment method. Deletion of a CRD means all the objects also get deleted, which means all the snapshots get deleted.

**Pull request closed without merging**